typing/mtype.ml

```diff
@@ -54,9 +54,8 @@ and strengthen_sig ~aliasable env sg p =
     [] -> []
   | (Sig_value(_, _, _) as sigelt) :: rem ->
       sigelt :: strengthen_sig ~aliasable env rem p
-  | Sig_type(id, {type_kind=Type_abstract}, _, _) ::
-    (Sig_type(id', {type_private=Private}, _, _) :: _ as rem)
-    when Ident.name id = Ident.name id' ^ "#row" ->
+  | Sig_type(id, {type_kind=Type_abstract}, _, _) :: rem
+    when Btype.is_row_name (Ident.name id) ->
```

Comment:
Code-review remark for myself: the present code would only drop row types when the corresponding type was Private, while the current code does not check this anymore (it could drop the "row type of a Public type"). But row types are only creates for private types anyhow, so there is no actual change of behavior here.

To check this: Look for `#row` in typedecl.ml for the two places that introduce these ghost type declarations. See that they only run when `is_fixed_type sdecl` holds. This check only passes for Private type declartions.